src/apps/angle/helpers/angle.api.ts

```diff
@@ -0,0 +1,171 @@
+
+import { ANGLE_DEFINITION } from '../angle.definition';
+
+const BASE_URL = 'https://api.angle.money/v1';
+
+type TAPR = {
+  value: number;
+  address: string;
+  details: {
+    min: number;
+    max: number;
+    fees: number;
+    interests: number;
+  };
+};
+
+type TokenInfo = {
+  readonly name: string;
+  readonly decimals: number;
+  readonly symbol: string;
+  readonly isSanToken?: boolean;
+  readonly useInSwap?: boolean;
+  readonly hasPermit?: boolean;
+  readonly permitVersion?: string;
+  readonly logoURI?: string;
+  readonly tags?: string[];
+};
+
+type TVaultManager = {
+  address: string;
+  borrowFee: number;
+  collateral: string;
+  collateralHasPermit: boolean;
+  collateralPermitVersion: string;
+  decimals: number;
+  dust: number;
+  liquidationPenalty: number;
+  maxLTV: number;
+  minCollateralRatio: number;
+  rate: number;
+  stabilityFee: number;
+  stablecoin: string;
+  swapper: string;
+  symbol: string;
+  totalCollateral: number;
+  totalDebt: number;
+  treasury: string;
+};
+
+type TPerpetual = {
+  perpetualID: string;
+  owner: string;
+  decimals: string;
+  margin: string;
+  committedAmount: string;
+  entryRate: string;
+  perpetualManager: string;
+  stableAddress: string;
+  collatAddress: string;
+  stableName: string;
+  collatName: string;
+  openingTimestamp: string;
+  lastUpdateTimestamp: string;
+  openingBlockNumber: string;
+  lastUpdateBlockNumber: string;
+  status: string;
+  claimable: number;
+};
+
+type TVault = {
+  address: string;
+  collateral: string;
+  collateralAmount: number;
+  collateralRatio: number;
+  debt: number;
+  debtString: string;
+  id: number;
+  liquidationPrice: number;
+  rate: number;
+  stablecoin: string;
+  symbol: string;
+};
+
+@Injectable()
+export class AngleApiHelper {
+  constructor(@Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit) {}
+
+  async #callAngleApi<T>(endpoint: string, params?: Record<string, any>): Promise<T> {
+    let url = `${BASE_URL}/${endpoint}`;
+    if (params) {
+      url += `?${querystring.stringify(params)}`;
+    }
+    const data = await Axios.get<T>(url).then(v => v.data);
+    return data;
+  }
+
+  @Cache({
+    instance: 'business',
+    key: (network: Network) => `studio:${ANGLE_DEFINITION}:apr:${network}:angle`,
+    ttl: 15 * 60,
+  })
+  async getApr() {
+    return this.#callAngleApi<Record<string, TAPR>>('apr');
+  }
+
+  @Cache({
+    instance: 'business',
+    key: (network: Network) => `studio:${ANGLE_DEFINITION}:vaultmanagers:${network}:angle`,
+    ttl: 15 * 60,
+  })
+  async getVaultManagers() {
+    return this.#callAngleApi<Record<string, TVaultManager>>('vaultManagers', { chainId: 1 });
+  }
+
+  @Cache({
+    instance: 'user',
+    key: (network: Network) => `studio:${ANGLE_DEFINITION}:perpetuals:${network}:angle`,
+    ttl: 15 * 60,
+  })
+  async getUserPerpetuals(address: string) {
+    return this.#callAngleApi<{ perpetuals: TPerpetual[] }>('perpetuals', {
```

Comment:
Could you remove the `#` prefix?

Reply:
this is just to make the method private. is it an issue? i can definitely remove it if it's a problem.
I added it to make sure the method wouldn't be used outside, so that all calls to the API would be made inside AngleApiHelper and be cached properly

Reply:
@immasandwich

Reply:
You can simply make it private via:


```ts
private async callAngleApi<T>(endpoint: string, params?: Record<string, any>): Promise<T> {
  // ...
}
```

Reply:
ah ok, of course :) i just changed it